homeassistant/components/sonos/switch.py

```diff
@@ -35,15 +36,12 @@
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up Sonos from a config entry."""
 
-    configured_alarms = set()
-
-    async def _async_create_entity(speaker: SonosSpeaker, new_alarms: set) -> None:
-        for alarm_id in new_alarms:
-            if alarm_id not in configured_alarms:
-                _LOGGER.debug("Creating alarm with id %s", alarm_id)
-                entity = SonosAlarmEntity(alarm_id, speaker)
-                async_add_entities([entity])
-                configured_alarms.add(alarm_id)
+    async def _async_create_entity(speaker: SonosSpeaker, alarms: list[Alarm]) -> None:
+        entities = []
+        for alarm in alarms:
+            _LOGGER.debug("Creating alarm %s on %s", alarm.alarm_id, speaker.zone_name)
+            entities.append(SonosAlarmEntity(alarm.alarm_id, speaker))
+        async_add_entities(entities)
```

Comment:
I think something like this could work, if we wrap this line in a for loop: https://github.com/home-assistant/core/blob/264893c46c66004f7f0f2f38a070c6a09f81a25c/homeassistant/components/sonos/speaker.py#L227
 
```suggestion
    async def _async_create_entity(speaker: SonosSpeaker, alarm_id: str) -> None:
        _LOGGER.debug("Creating alarm %s on %s", alarm.alarm_id, speaker.zone_name)
        async_add_entities([SonosAlarmEntity(alarm_id, speaker)])
```

Reply:
The current way will create all of the alarms for a given speaker in one call to `async_add_entities`. I think that's more efficient than dispatching repeatedly.

Reply:
Yes I agree!